incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/stream/OpenapiServerFactory.java

```diff
@@ -0,0 +1,1433 @@
+    private final HttpEndExFW.Builder httpEndExRW = new HttpEndExFW.Builder();
+    private final HttpResetExFW.Builder httpResetExRW = new HttpResetExFW.Builder();
+
+    private final MutableDirectBuffer writeBuffer;
+    private final MutableDirectBuffer extBuffer;
+    private final MutableDirectBuffer metadataBuffer;
+    private final BufferPool bufferPool;
+    private final Signaler signaler;
+    private final BindingHandler streamFactory;
+    private final LongUnaryOperator supplyInitialId;
+    private final LongUnaryOperator supplyReplyId;
+    private final LongSupplier supplyTraceId;
+    private final Long2ObjectHashMap<OpenapiBindingConfig> bindings;
+    private final int grpcTypeId;
+    private final int httpTypeId;
+
+
+    public OpenapiServerFactory(
+        OpenapiConfiguration config,
+        EngineContext context)
+    {
+        this.writeBuffer = context.writeBuffer();
+        this.extBuffer = new UnsafeBuffer(new byte[writeBuffer.capacity()]);
+        this.metadataBuffer = new UnsafeBuffer(new byte[writeBuffer.capacity()]);
+        this.bufferPool = context.bufferPool();
+        this.signaler = context.signaler();
+        this.streamFactory = context.streamFactory();
+        this.supplyInitialId = context::supplyInitialId;
+        this.supplyReplyId = context::supplyReplyId;
+        this.supplyTraceId = context::supplyTraceId;
+        this.bindings = new Long2ObjectHashMap<>();
+        this.grpcTypeId = context.supplyTypeId(OpenapiBinding.NAME);
+        this.httpTypeId = context.supplyTypeId(HTTP_TYPE_NAME);
+    }
+
+    @Override
+    public int originTypeId()
+    {
+        return httpTypeId;
+    }
+
+    @Override
+    public int routedTypeId()
+    {
+        return grpcTypeId;
+    }
+
+    @Override
+    public void attach(
+        BindingConfig binding)
+    {
+        OpenapiBindingConfig grpcBinding = new OpenapiBindingConfig(binding, metadataBuffer);
+        bindings.put(binding.id, grpcBinding);
+    }
+
+    @Override
+    public void detach(
+        long bindingId)
+    {
+        bindings.remove(bindingId);
+    }
+
+    @Override
+    public MessageConsumer newStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length,
+        MessageConsumer network)
+    {
+        final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+        final long originId = begin.originId();
+        final long routedId = begin.routedId();
+        final long traceId = begin.traceId();
+        final long authorization = begin.authorization();
+        final long initialId = begin.streamId();
+        final long sequence = begin.sequence();
+        final long acknowledge = begin.acknowledge();
+        final OctetsFW extension = begin.extension();
+        final HttpBeginExFW httpBeginEx = extension.get(httpBeginExRO::tryWrap);
+
+        MessageConsumer newStream = (t, b, i, l) -> {};
+
+        if (!isGrpcRequestMethod(httpBeginEx))
```

Comment:
This is where we need to check for streams coming from this bindings composite namespace(s) only and reject others.